### PR TITLE
Add Hoon

### PR DESCRIPTION
**Pull request by ilyakooo0**
reopening #9024

### CODE REVIEW

**Review by the-mikedavis — COMMENTED** [review]
It looks like the highlights queries have disappeared (`runtime/queries/hoon/highlights.scm`)

**Review by the-mikedavis — COMMENTED** [review]
To fix the docs CI you'll need to run `cargo xtask docgen` and commit the changes

File: runtime/queries/hoon/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(number) @number
```
Comment (the-mikedavis) [inline runtime/queries/hoon/highlights.scm:1]:
```suggestion
(number) @constant.numeric
```

**Review by the-mikedavis — DISMISSED** [review]

**Comment by pascalkuthe** [comment]
Needs to be rebas3d before merge

**Review by the-mikedavis — DISMISSED** [review]

**Review by pascalkuthe — DISMISSED** [review]

**Comment by pascalkuthe** [comment]
Oh looks like the comment about docgen wasn't addressed yet

**Comment by the-mikedavis** [comment]
Ah actually that's my bad, I just mixed up the docs changes in the merge commit

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**